Dapper.Contrib/SqlMapperExtensions.cs

```diff
@@ -657,19 +702,58 @@ public int Insert(IDbConnection connection, IDbTransaction transaction, int? com
     {
         var cmd = String.Format("insert into {0} ({1}) values ({2})", tableName, columnList, parameterList);
         connection.Execute(cmd, entityToInsert, transaction, commandTimeout);
-        var r = connection.Query("select @@IDENTITY id", transaction: transaction, commandTimeout: commandTimeout);
+        var r = connection.Query("select @@IDENTITY id", transaction: transaction, commandTimeout: commandTimeout).ToList();
+
+        if (r.First().id == null) return 0;
+        var id = (int) r.First().id;
+
+        var pi = keyProperties as PropertyInfo[] ?? keyProperties.ToArray();
+        if (!pi.Any()) return id;
+
+        var idp = pi.First();
+        idp.SetValue(entityToInsert, Convert.ChangeType(id, idp.PropertyType), null);
+
+        return id;
+    }
+
+    public void AppendColumnName(StringBuilder sb, string columnName)
+    {
+        sb.AppendFormat("[{0}]", columnName);
+    }
+
+    public void AppendColumnNameEqualsValue(StringBuilder sb, string columnName)
+    {
+        sb.AppendFormat("[{0}] = @{1}", columnName, columnName);
+    }
+}
+
+public partial class MySqlAdapter : ISqlAdapter
+{
+    public int Insert(IDbConnection connection, IDbTransaction transaction, int? commandTimeout, String tableName, string columnList, string parameterList, IEnumerable<PropertyInfo> keyProperties, object entityToInsert)
+    {
+        var cmd = String.Format("insert into {0} ({1}) values ({2})", tableName, columnList, parameterList);
+        connection.Execute(cmd, entityToInsert, transaction, commandTimeout);
+        var r = connection.Query("Select LAST_INSERT_ID()", transaction: transaction, commandTimeout: commandTimeout);

```

Comment:
I got the time to try this out, thank you again for your work! Though looks like I'm late it was already merged into master.

You got the last_insert_id() right for MySQL, instead of last_insert_rowid()
I ran Dapper.Contrib.Tests, and the ShortIdentity() test failed here:
```
public void ShortIdentity()
        {
            using (var connection = GetOpenConnection())
            {
                var id = connection.Insert(new Stuff() { Name = "First item" });
                id.IsEqualTo(1);
```
id was 0, so it was not equal to 1 as expected.

I just followed the lead of the SQLiteAdapter.Insert method in changing the above to:
```
var cmd = String.Format("insert into {0} ({1}) values ({2}); select last_insert_id() id", tableName, columnList, parameterList);
var r = connection.QueryMultiple(cmd, entityToInsert, transaction, commandTimeout);
var id = (int)r.Read().First().id; //(int) cast necessary as MySQL returns last_insert_id() as ulong.
//if (id == null) return 0; //this line no longer necessary.
```
With that change all tests passed when using MySQL.